src/Sarif.Multitool/AbsoluteUriCommand.cs

```diff
@@ -36,6 +36,12 @@ public int Run(AbsoluteUriOptions absoluteUriOptions)
 
                     // Write output to file.
                     string outputName = sarifLog.GetOutputFileName(absoluteUriOptions);
+
+                    // This isn't optimal. If there are several input files, and the first few output files
+                    // don't exist but the next one does, we will successfully process the first few files
+                    // and then fail. It's better than overwriting the existing file, though.
+                    DriverUtilities.VerifyOutputFileCanBeCreated(outputName, absoluteUriOptions.Force, _fileSystem);
```

Comment:
>DriverUtilities [](start = 20, length = 15)

Can this be done in GetSarifFiles() so we fail earlier?

Reply:
Yes, but it will take a bit of code rearrangement. `GetSarifFiles` creates a set of `AbsoluteUriFile` objects, each of which contains one of the input file paths. As the code stands, it's not until `GetOutputFileName` tells you the name of one of the corresponding output files that you know what you're looking for.

But there's no reason that an `AbsoluteUriFile` object can't be contain a property `OutputFileName`, and `GetSarifFiles` could certainly call `GetOutputFileName` to populate it for each `AbsoluteUriFile` object it returns.

TL;DR: Good idea, I'll fix it.

---
In reply to: [311713986](https://github.com/microsoft/sarif-sdk/pull/1625#discussion_r311713986) [](ancestors = 311713986)